systemtests/src/main/java/io/enmasse/systemtest/messaginginfra/ResourceManager.java

```diff
@@ -139,96 +130,69 @@ public MqttClientFactory getMqttClientFactory() {
     // Common resource methods
     //------------------------------------------------------------------------------------------------
 
-    public <T extends HasMetadata> void createResource(T... resources) {
-        for (T resource : resources) {
-            LOGGER.info("Create/Update of {} {} in namespace {}",
-                    resource.getKind(), resource.getMetadata().getName(), resource.getMetadata().getNamespace() == null ? "(not set)" : resource.getMetadata().getNamespace());
-            switch (resource.getKind()) {
-                case MessagingInfraCrd.KIND:
-                    createMessagingInfra((MessagingInfra) resource);
-                    scheduleDeletion(MessagingInfraCrd.getClient(), (MessagingInfra) resource);
-                    waitForInfraUp((MessagingInfra) resource);
-                    break;
-                default:
-                    LOGGER.warn("Can't find resource in list, please create it manually");
-                    break;
+    private final ResourceType<?>[] resourceTypes = new ResourceType[]{
+            new MessagingInfraResourceType(),
+            new MessagingTenantResourceType(),
+    };
+
+    @SafeVarargs
+    public final <T extends HasMetadata> void createResource(T... resources) {
```

Comment:
@lulf what about change behavior little bit -> for each resource type; do create; schedule deletion; end and then for each resource type; do wait for resource ready; end

Reply:
so then we can call create resources with for example infra and tenant and addresses in one command at same time